### PR TITLE
Added support for http-connect on the client side.

### PR DESCRIPTION
**Pull request by cheftako**
Add support for http-connect on client to server communication.
Uses --mode=http-connect on the server.
Client can be simulated using 
curl -v -p --proxy-key certs/master/private/proxy-client.key --proxy-cert certs/master/issued/proxy-client.crt --proxy-cacert certs/master/issued/ca.crt --proxy-cert-type PEM -x https://127.0.0.1:8090  http://localhost:8000

Also adds customizable ports.

### CODE REVIEW

File: cmd/proxy/main.go [hunk 1]
```diff
@@ -142,16 +186,19 @@ func (p *Proxy) run(o *ProxyRunOptions) error {
 	}
 	server := agentserver.NewProxyServer()
 
+	glog.Warning("Starting master server for client connections.")
```
Comment (anfernee) [inline cmd/proxy/main.go]: Warning => Info

File: pkg/agent/agentserver/tunnel.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	"math/rand"
+	"github.com/golang/glog"
+)
+
+type Tunnel struct {
+	Server *ProxyServer
+}
+
+func (t *Tunnel) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	log.Printf("Received %s request to %q from %v",
```
Comment (anfernee) [inline pkg/agent/agentserver/tunnel.go]: log=>glog

File: pkg/agent/agentserver/tunnel.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+		glog.Errorf("context reports done!!!")
+	default:
+	}
+	select {
+	case <-connection.connected: // Waiting for response before we begin full communication.
+	}
+	
+	defer conn.Close()
+
+	glog.Warningf("Starting proxy to %q", r.Host)
```
Comment (anfernee) [inline pkg/agent/agentserver/tunnel.go]: ditto. warning to info

**Review by anfernee — COMMENTED** [review]
minor: some log change.

File: cmd/proxy/main.go (hunk 1 above)
Comment (cheftako) [reply cmd/proxy/main.go]: Done

File: pkg/agent/agentserver/tunnel.go (hunk 2 above)
Comment (cheftako) [reply pkg/agent/agentserver/tunnel.go]: Good catch.

**Comment by anfernee** [comment]
thanks @cheftako !

**Pull request merged**